### PR TITLE
Mongodb support

### PR DESCRIPTION
**Pull request by mark-kar**
(PR to https://github.com/rafafrdz/criteria4s/issues/14 issue)

Hi! I've added a mongodb interpreter in this PR. So I would be happy if you @rafafrdz look at it :)

I based on the official [mongodb docs](https://www.mongodb.com/docs/) writing it, but I guess the future test suite will cover it in a more reliable way

### CODE REVIEW

File: mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/MongoDb.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package io.github.rafafrdz.criteria4s.mongodb
+
+import io.github.rafafrdz.criteria4s.core.Conjuction.{AND, OR}
+import io.github.rafafrdz.criteria4s.core.{Column, CriteriaTag, Show}
+import io.github.rafafrdz.criteria4s.core.PredicateBinary._
+import io.github.rafafrdz.criteria4s.core.PredicateUnary._
+import io.github.rafafrdz.criteria4s.instances._
+
+trait MongoDb extends CriteriaTag
```
Comment (rafafrdz) [inline mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/MongoDb.scala:9]:
```suggestion
trait MongoDB extends CriteriaTag
```

File: mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/MongoDb.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  private val betweenExample: (String, String) => String =
+    (ref, range) => s"{$ref: $range}"
+
+  implicit val showColumn: Show[Column, MongoDb] = Show.create(col => s"\"${col.colName}\"")
+  implicit def showSeq[T](implicit show: Show[T, MongoDb]): Show[Seq[T], MongoDb] =
+    Show.create(_.map(show.show).mkString("[", ", ", "]"))
+  implicit def betweenShow[T](implicit show: Show[T, MongoDb]): Show[(T, T), MongoDb] =
+    Show.create { case (l, r) => s"{ $$gte: ${show.show(l)}, $$lt: ${show.show(r)} }" }
+
+  trait MongoDbExpr[T <: MongoDb] {
```
Comment (rafafrdz) [inline mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/MongoDb.scala:36]:
```suggestion
  trait MongoDBExpr[T <: MongoDB] {
```

File: mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/package.scala [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package io.github.rafafrdz.criteria4s
```
Comment (rafafrdz) [inline mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/package.scala:1]:
```suggestion
package io.github.rafafrdz.criteria4s.dialect
```

**Review by rafafrdz — CHANGES_REQUESTED** [review]
Great! I like it. Just few suggestions:

* I know that scala suggests camel-case syntax as best-practice but I would like to name the mongodb dialect `MongoDB` instead of `MongoDb`

* Also, I was thinking about move all dialects like mongodb, postgresql and so on in a new sub-level in the package called `dialect`. I opened a PR (#25 ) to do that for SQL dialect.

With that, people will get any "dialect" under `io.github.rafafrdz.criteria4s.dialect` instead of   `io.github.rafafrdz.criteria4s` which contains also the core, functions and so on.

```scala
// before
io.github.rafafrdz.criteria4s.mongodb._
io.github.rafafrdz.criteria4s.postgresql._

// now
io.github.rafafrdz.criteria4s.dialect.mongodb._
io.github.rafafrdz.criteria4s.dialect.postgresql._
```

What do you think? your opinion is important for me :)

File: mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/MongoDb.scala (hunk 1 above)
Comment (mark-kar) [reply mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/MongoDb.scala:9]: Thanks for reacting! I really appreciate your comments :)

1. I agree. I guess it also doesn't hurt the scala codestyle as this naming is specific for MongoDB.
2. I like it. Thus we separate our interpreters from the algebra I think. Good idea!

So then, can you merge firstly your https://github.com/rafafrdz/criteria4s/pull/25 PR? Then I'll make the mentioned corrections and merge with your changes properly

Comment (rafafrdz) [reply mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/MongoDb.scala:9]: Perfect!

> So then, can you merge firstly your https://github.com/rafafrdz/criteria4s/pull/25 PR? Then I'll make the mentioned corrections and merge with your changes properly

approve me it (#25) please 🤖

Comment (mark-kar) [reply mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/MongoDb.scala]: Ready😉

File: mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/package.scala (hunk 3 above)
Comment (rafafrdz) [reply mongodb/src/main/scala/io/github/rafafrdz/criteria4s/mongodb/package.scala:1]: Just the package and go ahead! 🚀

**Review by rafafrdz — APPROVED** [review]
Blessed 🙏🏼

**Pull request merged**